### ISSUE TITLE
terra-date-picker does not highlight full text on mobile iOS devices

### ISSUE THREAD

**Issue by kw070877** [cerner/terra-framework#1494]
# Bug Report

## Description

On mobile iOS devices, when selecting into the date input, the input is only partially selected (see screenshot) due to the keyboard animation.

## Steps to Reproduce

1. Load a terra-date-picker on an iOS mobile device or simulator (with the soft keyboard on)
2. Click on any input (month, day, year)
3. Text is only partially selected

## Additional Context / Screenshots

<img width="363" alt="Screen Shot 2021-12-17 at 8 33 27 AM" src="https://user-images.githubusercontent.com/85512188/146560747-a28f31b1-2fd4-4afd-8823-487076f0da11.png">

## Expected Behavior

Full text should be highlighted.

## Possible Solution

I have tested that this can be fixed by wrapping the [setSelectionRange](https://github.com/cerner/terra-framework/blob/main/packages/terra-date-picker/src/DateInput.jsx#L533) calls in a `setTimeout(() => ..., 0):` https://stackoverflow.com/a/7436574

## Environment

* Component Name and Version: terra-date-picker 4.71.0
* Browser Name and Version: Mobile Safari
* Node/npm Version: lts/dubnium
* Webpack Version: webpack 4.41.2
* Operating System and version (desktop or mobile): iOS 15

**Comment by neilpfeiffer** [comment]
@kw070877 It sounds like from your issue description that you may be expecting the full value, e.g. `07/20/2016`, to be selected when the user selects any of the individual input nodes (month, day, or year)? Instead of the selection of just that particular input node value, e.g. just `2016`?

If so, it did work like this previously prior to [version 4.69.0](https://engineering.cerner.com/terra-ui/components/terra-date-picker/date-picker/change-log#4690---november-2-2021), which is when we introduced splitting the month, day, and year inputs into actual separated html inputs for better interaction reliability and validation ([PR #1208](https://github.com/cerner/terra-framework/pull/1208).  But the problems with the previous version with the numbers+text being all one actual value was a user had to correctly enter the ` / ` separators to form a valid date, which was a large dissatisfier, and any additional JS mask corrections to auto-handle including the separators were unstable across various devices and browsers.

Testing the current component v4.71.1, it was performing as designed of selecting the full value of a single node (e.g. just the month, just the day, or just the year) when trying to select it, and was unable to replicate it selecting less than the full four digits of the year as your screenshot shows (e.g. just `16`), or less than the full month or full day value.

**Comment by kw070877** [comment]
@neilpfeiffer I am not talking about the selection of the full date but rather the selection of the full month/date/year. When selecting the year in the above screenshot on an iOS device, I expected the full year to get highlighted '2016', however, only the '16' got highlighted.

**Comment by neilpfeiffer** [comment]
@kw070877 What specific environment details are you producing the issue (only simulator or on actual devices, web directly or in iOS shell app webview)? And what current behavior are you seeing when the month and day inputs are being selected (without the addition of the setTimeout delay)?

**Comment by kw070877** [comment]
@neilpfeiffer This can be reproduced on a simulator (only with the physical keyboard enabled) and on actual devices. It happens in both safari/webview directly and iOS app webview. The behavior is the inputs are not fully selected. 

Are you not able to reproduce this on an iOS device/simulator? I can certainly get a gif/video if needed to demonstrate the behavior.

**Comment by neilpfeiffer** [comment]
No, I am not able to recreate it on 3 different iOS devices running different major iOS versions, using our [terra-ui](https://engineering.cerner.com/terra-ui/components/terra-date-picker/date-picker/date-picker) site (v4.69.1) or the [terra-framework staging](https://engineering.cerner.com/terra-framework/components/terra-date-picker/date-picker/date-picker) site page (v4.71.1) for date-picker, I have not tried the iOS simulator yet, so information about what environments you have this being produced will be helpful.

**Comment by benbcai** [comment]
I tried using chrome and safari on iOS and also unable to recreate. 

<img width="213" alt="Screen Shot 2022-01-06 at 4 30 14 PM" src="https://user-images.githubusercontent.com/22618655/148461441-0764e292-cfca-4640-9d63-b81a66bfff69.png">

@kw070877 Can you try using one of the examples on [terra-ui](https://engineering.cerner.com/terra-ui/components/terra-date-picker/date-picker/date-picker) to see if you can recreate?

**Comment by neilpfeiffer** [comment]
After further testing with @kw070877, we were able to determine the problem was reproducible using their project, as well as the date-picker doc site pages on terra-ui, but only when using the Apple iOS simulator on Mac. 

We however were not able to reproduce the same behavior effects on actual iOS devices using the same test samples, and after some research, setSelectionRange was historically flaky in the past with mobile Safari, but has been resolved for some years. We determined since this doesn't seem to be an issue anywhere on actual devices, and since no scripted testing relies on using the iOS simulator, we will close this issue for now and can re-open if it is ever observed on user devices. Without extensive investigation as to why the simulator might still be having residual issues and actual devices do not, this issue should be fine to resolve as not a critical bug or issue.

cc: @benbcai
